src/configuration.cpp

```diff
@@ -172,6 +179,48 @@ chimera::CompiledConfiguration::CompiledConfiguration(
   , bindingNode_(configNode_["template"]) // TODO: is this always ok?
   , ci_(ci)
 {
+    // This placeholder will be filled in by the options.strict specified
+    // in the configuration YAML if it exists, or remain unset otherwise.
+    boost::optional<bool> config_options_strict;
+
+    // Parse 'options' section of configuration YAML if it exists.
+    if (configNode_)
+    {
+        const YAML::Node &optionsNode = configNode_["options"];
+        if (optionsNode)
+        {
+            const YAML::Node &strictNode = optionsNode["strict"];
+            if (strictNode)
+            {
+                // Check that 'strict' node in configuration YAML is a scalar.
+                if (!strictNode.IsScalar())
+                {
+                    std::cerr << "'options.strict' in configuration YAML must "
+                                 "be a scalar."
+                              << std::endl;
+                    exit(-2);
+                }
+                config_options_strict = strictNode.as<bool>();
+            }
+        }
+    }
+
+    // Set the options.strict from one of the following sources in order of
+    // priority: 1) CLI '-strict' setting (True if -strict passed, False
+    // otherwise) 2) YAML configuration setting (True/False) 3) false by default
+    if (parent.strict_)
+    {
+        strict_ = parent.strict_;
+    }
+    else if (config_options_strict)
+    {
+        strict_ = *config_options_strict;
```

Comment:
If you can do optional chaining, then the YAML logic can probably just go in here, since it will be much shorter.

Reply:
We still need to have the YAML logic out of this if-statement because we should check if (1) the tag is specified and (2) the YAML node is a scalar type before this if-statement.